bin/agitator

```diff
@@ -30,6 +30,81 @@ Possible commands:
 EOF
 }
 
+# Starts a app specific agitator
+# usage: start_app_agitator app_name kill_sleep_time restart_sleep_time min_kill max_kill start_cmd kill_cmd
+# Requires that a list of hosts to be set in $APP_HOSTS
+function start_app_agitator() {
+  local app_name=$1
+  local kill_sleep_time=$2
+  local restart_sleep_time=$3
+  local min_kill=$4
+  local max_kill=$5
+  local start_cmd=$6
+  local kill_cmd=$7
+  local hosts_array
+  get_app_hosts "$app_name" hosts_array
```

Comment:
The use of a nameref in `get_app_hosts` is clever, but I think it's more complicated than necessary. It's much simpler to `echo` the hosts in `get_app_hosts` and simply capture that output into an array here:

```bash
  local hosts_array; hosts_array=($(get_app_hosts "$app_name"))
```

This would be similar to Java:
```java
String[] hostsArray = getAppHosts(appName).split(" ");
```

Shellcheck won't like not quoting the output, even though that's what you want here. So the best way is to explicitly specify the delimiter when constructing the array, using `mapfile` (aka `readarray`):

```suggestion
  local hosts_array; readarray -td' ' hosts_array < <(get_app_hosts "$app_name")
```

Reply:
This is neat. It also seems less obscure or hazardous then using `nameref`. I will use this technique.

I assume the space between the 2 `<` symbols was intentional?